packages/mc-scripts/src/utils/get-config-diff.js

```diff
@@ -0,0 +1,252 @@
+  });
+
+  return arrayDiff.join('\n');
+};
+
+const getPermissionsDiff = (oldPermissions, newPermissions) => {
+  const permissionDiff = ['permissions changed'];
+  const mappedOldPermission = {};
+
+  oldPermissions.forEach(({ name, oAuthScopes }) => {
+    mappedOldPermission[name] = oAuthScopes;
+  });
+
+  newPermissions.forEach((newPermission) => {
+    const currDiff = [`${t(1)}${newPermission.name} changed`];
+    // if the permission name is not in the old config, it means it is a new addition.
+    if (!mappedOldPermission[newPermission.name]) {
+      permissionDiff.push(`${t(1)}"${green(newPermission.name)}" was added`);
+    }
+
+    // if permission name is in the old config, now we check if there was a change
+    else {
+      currDiff.push(
+        getArrayDiff(
+          mappedOldPermission[newPermission.name],
+          newPermission.oAuthScopes,
+          2
+        )
+      );
+      delete mappedOldPermission[newPermission.name];
+    }
+
+    currDiff.filter(Boolean).length > 1 &&
+      permissionDiff.push(currDiff.join('\n'));
+  });
+
+  // if there are old permissions left, it means they were deleted in the new Permissions.
+  Object.keys(mappedOldPermission).forEach((oldPermissionName) => {
+    permissionDiff.push(`${t(1)}"${red(oldPermissionName)}" was removed`);
+  });
+
+  if (permissionDiff.length > 1) return permissionDiff.join('\n');
+};
+
+const getLabelAllLocalesDiff = (
+  oldLabelAllLocales,
+  newLabelAllLocales,
+  indentLevel = 0
+) => {
+  const mappedOldLabelAllLocales = {};
+  const labelAllLocalesDiff = [`${t(indentLevel - 1)}labelAllLocales changed`];
+
+  oldLabelAllLocales?.forEach(({ locale, value }) => {
+    mappedOldLabelAllLocales[locale] = value;
+  });
+
+  newLabelAllLocales?.forEach((newLabelAllLocale) => {
+    if (newLabelAllLocale.locale in mappedOldLabelAllLocales) {
+      const oldLocaleValue = mappedOldLabelAllLocales[newLabelAllLocale.locale];
+      if (oldLocaleValue !== newLabelAllLocale.value) {
+        labelAllLocalesDiff.push(
+          `${t(indentLevel)}locale with "${
+            newLabelAllLocale.locale
+          }" was changed: ${red(oldLocaleValue)} => ${green(
+            newLabelAllLocale.value
+          )}`
+        );
+      }
+      delete mappedOldLabelAllLocales[newLabelAllLocale.locale];
+    } else {
+      labelAllLocalesDiff.push(
+        `${t(indentLevel)}locale with "${green(
+          newLabelAllLocale.locale
+        )}" was added`
+      );
+    }
+  });
+
+  Object.keys(mappedOldLabelAllLocales).forEach((key) => {
+    labelAllLocalesDiff.push(
+      `${t(indentLevel)}locale with "${red(key)}" was removed`
+    );
+  });
+
+  if (labelAllLocalesDiff.length > 1) return labelAllLocalesDiff.join('\n');
+};
+
+const getMainMenuLinkDiff = (oldMainMenuLink, newMainMenuLink) => {
+  const mainMenuLinkDiff = ['mainMenuLink changed'];
+
+  mainMenuLinkDiff.push(
+    getStringDiff(
+      oldMainMenuLink['defaultLabel'],
```

Comment:
I'm curious if we have any linting rule that enforces using square brackets for accessing object properties (`oldMainMenuLink['defaultLabel']`) instead of dots (`oldMainMenuLink.defaultLabel`).